signature-aggregator/aggregator/aggregator.go

```diff
@@ -499,15 +502,21 @@ func (s *SignatureAggregator) CreateSignedMessage(
 	return signedMsg, nil
 }
 
-func (s *SignatureAggregator) getSubnetID(log logging.Logger, blockchainID ids.ID) (ids.ID, error) {
+func (s *SignatureAggregator) getSubnetID(
+	ctx context.Context,
+	log logging.Logger,
+	blockchainID ids.ID,
+) (ids.ID, error) {
 	s.subnetsMapLock.RLock()
 	subnetID, ok := s.subnetIDsByBlockchainID[blockchainID]
 	s.subnetsMapLock.RUnlock()
 	if ok {
 		return subnetID, nil
 	}
 	log.Info("Signing subnet not found, requesting from PChain", zap.String("blockchainID", blockchainID.String()))
-	subnetID, err := s.network.GetSubnetID(blockchainID)
+	ctx, cancel := context.WithTimeout(ctx, pChainSubnetIDTimeout)
```

Comment:
Nit: Should we name this `getSubnetIDCtx` so it's not overwriting the `ctx` passed in as a param? No functional difference, but it's error prone with we re-use ctx again below in the future as is.